modules/effects/arrow-effects/src/main/kotlin/arrow/effects/internal/Promise.kt

```diff
@@ -0,0 +1,104 @@
+package arrow.effects.internal
+
+import arrow.Kind
+import arrow.core.*
+import arrow.effects.Promise
+import java.util.concurrent.atomic.AtomicReference
+
+fun <A> Promise.Companion.unsafe(): Promise<ForId, A> = UnsafePromise()
+
+internal class UnsafePromise<A> : Promise<ForId, A> {
+
+  private val state: AtomicReference<State<A>> = AtomicReference(State.Pending(emptyList()))
+
+  override val get: Id<A>
```

Comment:
If we're overriding the get, might as well make it if a function. I know `def` acts to define both values and functions in Scala, but here it's better to be explicit.